crates/ruff_python_semantic/src/model.rs

```diff
@@ -546,6 +550,35 @@ impl<'a> SemanticModel<'a> {
         self.stmts.ancestor_ids(node_id).map(|id| self.stmts[id])
     }
 
+    /// Set the [`Globals`] for the current [`Scope`].
+    pub fn set_globals(&mut self, globals: Globals<'a>) {
+        // If any global bindings don't already exist in the global scope, add them.
+        for (name, range) in globals.iter() {
+            if self.global_scope().get(name).map_or(true, |binding_id| {
+                self.bindings[binding_id].kind.is_annotation()
+            }) {
+                let id = self.bindings.push(Binding {
+                    kind: BindingKind::Assignment,
+                    range: *range,
+                    references: Vec::new(),
+                    source: self.stmt_id,
+                    context: self.execution_context(),
+                    exceptions: self.exceptions(),
+                    flags: BindingFlags::empty(),
+                });
+                self.global_scope_mut().add(name, id);
+            }
+        }
+
+        self.scopes[self.scope_id].set_globals_id(self.globals.push(globals));
+    }
+
+    /// Return the [`TextRange`] at which a name is declared as global in the current [`Scope`].
+    pub fn get_global(&self, name: &str) -> Option<TextRange> {
```

Comment:
Nit: rename to 'global'? The rust convention is to prefix the setter with set but use the name alone for the getter